### PR TITLE
update for cedar#800

### PR DESCRIPTION
**Pull request by cdisselkoen**
*Issue #, if available:*

cedar-policy/cedar#800

### CODE REVIEW

File: CedarJava/src/main/java/com/cedarpolicy/model/DetailedError.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package com.cedarpolicy.model;
```
Comment (aaronjeline) [inline CedarJava/src/main/java/com/cedarpolicy/model/DetailedError.java:1]: Do we want to drop an explicit note that this is an encoding of `miette::Report`?

**Review by aaronjeline — APPROVED** [review]

File: CedarJava/src/main/java/com/cedarpolicy/model/DetailedError.java (hunk 1 above)
Comment (cdisselkoen) [reply CedarJava/src/main/java/com/cedarpolicy/model/DetailedError.java:1]: I think that's a Rust implementation detail.  As far as the Java is concerned, this is just the structure we get over JSON from Rust.  Having that comment on the Rust side is good though

File: CedarJava/src/main/java/com/cedarpolicy/model/AuthorizationSuccessResponse.java [hunk 2]
```diff
@@ -95,6 +95,29 @@ public AuthorizationSuccessResponse(
         this.diagnostics = diagnostics;
     }
 
+    /** Error (or warning) which occurred in a particular policy during authorization */
```
Comment (khieta) [inline CedarJava/src/main/java/com/cedarpolicy/model/AuthorizationSuccessResponse.java:98]: Nit: Currently there are no warnings returned at authorization time. Warnings would instead be captured in the `AuthorizationResponse` struct.

File: CedarJava/src/main/java/com/cedarpolicy/model/ValidationResponse.java [hunk 3]
```diff
@@ -28,62 +28,87 @@
 
 /** Result of a validation request. */
 public final class ValidationResponse {
-    /** Exactly one of `results` or `errors` should be None. */
+    /** Is this a success or a failure response */
+    @JsonProperty("type")
+    public final SuccessOrFailure type;
+    /** This will be present if and only if `type` is `Success`. */
     public final Optional<ValidationResults> results;
```
Comment (khieta) [inline CedarJava/src/main/java/com/cedarpolicy/model/ValidationResponse.java:35]: Nit: To match the naming in AuthorizationReponse.java
```suggestion
    public final Optional<ValidationSuccessResponse> success;
```

**Review by khieta — APPROVED** [review]
Left some non-blocking comments.

File: CedarJava/src/main/java/com/cedarpolicy/model/ValidationResponse.java (hunk 3 above)
Comment (cdisselkoen) [reply CedarJava/src/main/java/com/cedarpolicy/model/ValidationResponse.java]: Renamed and pushed

**Review by khieta — APPROVED** [review]

**Pull request merged**